pkg/destroy/openstack/swift.go

```diff
@@ -1,20 +1,43 @@
 package openstack
 
 import (
+	"time"
+
 	"github.com/gophercloud/gophercloud/openstack/objectstorage/v1/containers"
 	"github.com/gophercloud/gophercloud/openstack/objectstorage/v1/objects"
 	"github.com/gophercloud/utils/openstack/clientconfig"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 // DeleteSwiftContainer deletes a container and all of its objects.
 func DeleteSwiftContainer(name string, cloud string) error {
+	backoffSettings := wait.Backoff{
+		Duration: time.Second * 20,
+		Steps:    30,
+	}
+
+	err := wait.ExponentialBackoff(backoffSettings, func() (bool, error) {
+		return deleteSwiftContainer(name, cloud)
+	})
+	if err != nil {
+		return errors.Errorf("Unrecoverable error/timed out: %v", err)
+	}
+
+	return nil
+}
+
+func deleteSwiftContainer(name string, cloud string) (bool, error) {
 	opts := clientconfig.ClientOpts{
 		Cloud: cloud,
 	}
 
 	swiftClient, err := clientconfig.NewServiceClient("object-store", &opts)
 	if err != nil {
-		return err
+		logrus.Warningf("There was an error during the container removal: %v", err)
```

Comment:
All of these `warningf`  might be better suited with `debugf`. We normally don't show output if we expect a retry to succeed.

Reply:
done!